src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy

```diff
@@ -116,21 +116,38 @@ class DependencyManager {
             log.error("Unexpected exception during presenter setup.", e)
         }
         try {
-            SampleTrackingQueryDataSource trackingInfoCenter = SampleTracker.createSampleTrackingInformation(trackingServices.get(0), this.serviceUser)
+            SampleTrackingQueryDataSource trackingInfoCenter = SampleTracker.createSampleTrackingInformation(trackingServices.get(0), serviceUser)
             this.queryAvailableLocationsInteractor = new QueryAvailableLocations(trackingInfoCenter, controlElementsPresenter)
         } catch (Exception e) {
             log.error("Could not setup ${QueryAvailableLocations.getSimpleName()} use case", e)
         }
-
+        def userID = "not logged in"
         try {
-            SampleTrackingQueryDataSource trackingInfoCenter = SampleTracker.createSampleTrackingInformation(trackingServices.get(0), this.serviceUser)
-            this.querySampleInteractor = new QuerySample(trackingInfoCenter, sampleListPresenter)
+            userID = PortalUtils.getScreenName()
+        } catch (NullPointerException e) {
+            log.error("User not logged into Liferay. They won't be able to see samples.")
+        }
+        OpenBisClient openbis
+        Set<String> spaces = new HashSet<>()
+        try {
+            log.info("Trying to connect to openBIS")
+            openbis = new OpenBisClient(configManager.getDataSourceUser(), configManager.getDataSourcePassword(), configManager.getDataSourceUrl())
+            openbis.login()
+            log.info("Fetching user spaces for " + userID)
+            spaces.addAll(openbis.getUserSpaces(userID))
+        } catch (Exception e) {
+            Styles.notification("openBIS connection error",
```

Comment:
You could also just `log.error` the message and throw the Exception. The application then defaults to the default `Portlet currently not available` message which should suffice I think.

Reply:
thanks, missed this section!